internal/jobexecutor/job_executor.go

```diff
@@ -325,3 +325,25 @@ func (e *JobExecutor) reportError(ctx context.Context, res *jobExecutorResult) {
 		e.Logger.ErrorContext(ctx, e.Name+": Failed to report error for job", logAttrs...)
 	}
 }
+
+// captureStackTrace returns a formatted stack trace string starting after
+// skipping the specified number of frames. The skip parameter should be
+// adjusted so that frames you want to hide (like the ones generated by the
+// tracing functions themselves) are excluded.
+func captureStackTraceSkipFrames(skip int) string {
+	// Allocate room for up to 100 callers; adjust as needed.
+	pcs := make([]uintptr, 100)
+	// Skip the specified number of frames.
+	n := runtime.Callers(skip, pcs)
+	frames := runtime.CallersFrames(pcs[:n])
+
+	var stackTrace string
+	for {
+		frame, more := frames.Next()
+		stackTrace += fmt.Sprintf("%s\n\t%s:%d\n", frame.Function, frame.File, frame.Line)
```

Comment:
I assume the formatting comes from what `debug.Stack()` would be doing? A little unfortunate that we have to format it manually here.

Reply:
This is actually quite a bit easier to grok than the stdlib 😬 :

https://github.com/golang/go/blob/4267fd389e941cf197cc3890cc42e474866c0d30/src/runtime/traceback.go#L944-L1009

This is more similar to what pkg/errors does:

https://github.com/pkg/errors/blob/5dd12d0cfe7f152f80558d591504ce685299311e/stack.go#L68-L71